File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/rabbitmq/docs/properties/ProductClient.kt
package io.micronaut.rabbitmq.docs.properties

// tag::imports[]
import io.micronaut.rabbitmq.annotation.Binding
import io.micronaut.rabbitmq.annotation.RabbitClient
import io.micronaut.rabbitmq.annotation.RabbitProperties
import io.micronaut.rabbitmq.annotation.RabbitProperty
import io.micronaut.context.annotation.Requires
// end::imports[]

@Requires(property = "spec.name", value = "PropertiesSpec")
// tag::clazz[]
@RabbitClient
@RabbitProperty(name = "appId", value = "myApp") // <1>
interface ProductClient {

    @Binding("product")
    @RabbitProperties( // <2>
            RabbitProperty(name = "contentType", value = "application/json"),
            RabbitProperty(name = "userId", value = "guest")
    )
    fun send(data: ByteArray)

    @Binding("product")
    fun send(@RabbitProperty("userId") user: String, @RabbitProperty contentType: String?, data: ByteArray)  // <3>
}
// end::clazz[]

File: docs-examples/example-python/src/test/python/micronaut/rabbitmq/docs/properties/ProductClient.py
from abc import ABC, abstractmethod
from typing import Annotated

from micronaut.context.annotation import Requires
# tag::imports[]
from micronaut.rabbitmq.annotation import Binding, RabbitClient, RabbitProperty
# end::imports[]


@Requires(property="spec.name", value="PropertiesSpec")
# tag::clazz[]
@RabbitClient
@RabbitProperty(name="appId", value="myApp")  # <1>
@RabbitProperty(name="userId", value="admin")
class ProductClient(ABC):

    @Binding("product")
    @RabbitProperty(name="contentType", value="application/json")  # <2>
    @RabbitProperty(name="userId", value="guest")
    @abstractmethod
    def send(self, data: bytes) -> None:
        ...

    @Binding("product")
    @abstractmethod
    def send_with_properties(self, user: Annotated[str, RabbitProperty("userId")],
                             content_type: Annotated[str | None, RabbitProperty("contentType")],
                             data: bytes) -> None:  # <3>
        ...
# end::clazz[]
